src/MultiplayerMod/Multiplayer/UI/Notifications.cs

```diff
@@ -0,0 +1,27 @@
+using JetBrains.Annotations;
+using MultiplayerMod.Core.Dependency;
+using MultiplayerMod.Core.Events;
+using MultiplayerMod.Multiplayer.CoreOperations.Events;
+
+namespace MultiplayerMod.Multiplayer.UI;
+
+[Dependency, UsedImplicitly]
+public class Notifications {
+
+    public Notifications(EventDispatcher events) {
+        events.Subscribe<ConnectionToHostHasBeenLost>(OnConnectionLost);
+    }
+
+    private void OnConnectionLost(ConnectionToHostHasBeenLost @event) {
+        var screen = (InfoDialogScreen) GameScreenManager.Instance.StartScreen(
+            ScreenPrefabs.Instance.InfoDialogScreen.gameObject,
+            GameScreenManager.Instance.ssOverlayCanvas.gameObject
+        );
+        screen.SetHeader("Connection has been lost");
+        screen.AddPlainText("Connection has been lost. Further play can not be synced");
+        screen.AddOption(
+            "OK",
```

Comment:
I think it should be something more obvious, like "Quit", and maybe add an option to continue (and change game mode to single 🤔)?

Reply:
I'll change it to `Quit`.
I've tried to keep game running in single mode, however there are more corner cases that need to be fixed in order to proceed (not all bindings and components are being shut down upon GameQuitEvent ).
So even though it could be nice I'd like to keep continuation in a single mode out of a scope of the PR.